src/hashes/blake2b.c

```diff
@@ -342,7 +342,7 @@ int blake2b_process(hash_state *md, const unsigned char *in, unsigned long inlen
       unsigned long fill = BLAKE2B_BLOCKBYTES - left;
       if (inlen > fill) {
          md->blake2b.curlen = 0;
-         XMEMCPY(md->blake2b.buf + left, in, fill); /* Fill buffer */
+         XMEMCPY(md->blake2b.buf + (left % sizeof(md->blake2b.buf)), in, fill); /* Fill buffer */
```

Comment:
Well, we are obviously trying here to work around the situation when `left == BLAKE2B_BLOCKBYTES`. IMHO it would be better to avoid that instead of "fixing" it by %-trick.

BTW we should perhaps use `BLAKE2B_BLOCKBYTES` instead of  `sizeof(md->blake2b.buf)`